src/resultScanner.ts

```diff
@@ -0,0 +1,404 @@
+  'notMatch' : 'Current value must case-sensitive not match the target value.',
+  'matchInsensitively' : 'Current value must case-insensitive match the target value.',
+  'notMatchInsensitively' : 'Current value must case-insensitive not match the target value.'
+}
+
+function getPolicyEvaluationDetails(evalData : any) : any{
+  if(evalData == null || evalData == {}){
+    return "No Evaluation details received";
+  }
+  if(evalData.evaluatedExpressions == null || evalData.evaluatedExpressions.length == 0){
+    return "No expressions evaluated";
+  }
+  let finalVal : string = '{ ';
+  let index = 1;
+  evalData.evaluatedExpressions.forEach(element => {
+      if(index > 1) finalVal = finalVal + ',';
+      finalVal = finalVal + '\"'+ element.path + '\" : ' + JSON.stringify({
+          'REASON' : (CONDITION_MAP[element.operator.toString().toLowerCase()] ? CONDITION_MAP[element.operator.toString().toLowerCase()] : 'Not Parsed'),
+          'CurrentValue' : element.expressionValue,
+          'Condition' : element.operator,
+          'ExpectedValue' : JSON.stringify(element.targetValue).replace("[","(").replace("]",")")
+       });
+      index++;
+  });
+  finalVal = finalVal + ' }';
+  while(finalVal.indexOf('[') > -1 || finalVal.indexOf(']') > -1){
+    finalVal = finalVal.replace("[","(").replace("]",")");
+  }
+  //core.debug(`\nPolicyEvaluationDetails parsed: ${finalVal}`);
+  return JSON.parse(finalVal);
+}
+
+ export async function batchCall(batchUrl: string, batchMethod: string, batchRequests: any[], token: string): Promise<WebResponse> {
+  let batchWebRequest = new WebRequest();
+  batchWebRequest.method = batchMethod.length > 0 ? batchMethod :'POST';
+  batchWebRequest.uri = batchUrl.length > 0 ? batchUrl : `https://management.azure.com/batch?api-version=2020-06-01` ;
+  batchWebRequest.headers = {
+    'Authorization': `Bearer ${token}`,
+    'Content-Type': 'application/json; charset=utf-8'
+  }
+  batchWebRequest.body = batchRequests.length > 0 ? JSON.stringify({ 'requests' : batchRequests }) : ""; 
+
+  core.debug(`Batch request :: Batch URL: ${batchWebRequest.uri} # Requests: ${batchRequests.length}`);
+  if(batchRequests.length > 0){
+    core.debug(`\tRequest URL sample: => ${batchRequests[0].url}`);
+  }
+  return await sendRequest(batchWebRequest).then((response: WebResponse) => {
+    if (response.statusCode == 200 || response.statusCode == 202){
+      core.debug(`Batch response :: Status: ${response.statusCode} Location: ${response.headers['location']} Body: ${response.body}`);
+      return Promise.resolve( response );
+    }
+    return Promise.reject(`An error occured while fetching the batch result. StatusCode: ${response.statusCode}, Body: ${JSON.stringify(response.body)}`);
+  }).catch(error => {
+    return Promise.reject(error);
+  });
+ }
+
+  export async function batchCalls(uri: string, method: string ,commonHeaders: any, polls: any[], token: string): Promise<any[]> {    
```

Comment:
can this method be split into multiple smaller once?

Reply:
Fixed in [commit](https://github.com/Azure/policy-compliance-scan/pull/1/commits/af8c105b2b59b486be830afbe3511ea792a6bf6b?file-filters%5B%5D=.ts&file-filters%5B%5D=.yml)